test/e2e/scrapeconfig_test.go

```diff
@@ -2119,3 +2122,101 @@ var LightSailSDTestCases = []scrapeCRDTestCase{
 		expectedError: true,
 	},
 }
+
+var ScalewaySDTestCases = []scrapeCRDTestCase{
+	{
+		name: "Valid Zone value",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			ScalewaySDConfigs: []monitoringv1alpha1.ScalewaySDConfig{
+				{
+					ProjectID: "1",
+					Role:      monitoringv1alpha1.ScalewayRoleBaremetal,
+					AccessKey: "AccessKey",
+					Zone:      ptr.To("fr-par-1"),
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Invalid Zone value",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			ScalewaySDConfigs: []monitoringv1alpha1.ScalewaySDConfig{
+				{
+					Zone: ptr.To(""),
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Empty Zone value",
```

Comment:
This is a valid case and we would expect no errors, since `Zone` is an optional field. The test fails since all the required fields have not been configured and the object is rejected for that reason (not for the fact that `Zone` is unspecified).